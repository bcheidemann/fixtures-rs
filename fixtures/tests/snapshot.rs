use indoc::indoc;
use std::process::Command;

fn test_example_expansion(example_name: &str) {
    let output = Command::new("cargo")
        .args(["expand", "--tests", "--example", example_name])
        .output()
        .expect("failed to expand example");

    if !output.status.success() {
        let stdout = String::from_utf8_lossy(&output.stdout);
        let stderr = String::from_utf8_lossy(&output.stderr);
        panic!(
            indoc! {"
                error: failed to expand example ({})

                -- begin stdout --
                {}
                -- end stdout --

                -- begin stderr --
                {}
                -- end stderr --
            "},
            example_name, stdout, stderr
        );
    }

    let current_dir = std::env::current_dir()
        .expect("failed to get current directory")
        .to_string_lossy()
        .to_string();
    let expansion = String::from_utf8_lossy(&output.stdout).replace(&current_dir, "<repo>");

    insta::assert_snapshot!(example_name, expansion);
}

#[test]
fn basic_usage() {
    test_example_expansion("basic_usage");
}

#[test]
fn complex_signatures() {
    test_example_expansion("complex_signatures");
}

#[test]
fn criterion() {
    test_example_expansion("criterion");
}

#[test]
fn ignore_attributes() {
    test_example_expansion("ignore_attributes");
}

#[test]
fn multiple_fixtures() {
    test_example_expansion("multiple_fixtures");
}

#[test]
fn negative_globs() {
    test_example_expansion("negative_globs");
}
