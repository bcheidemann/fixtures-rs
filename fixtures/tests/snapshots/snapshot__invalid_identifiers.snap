---
source: fixtures/tests/snapshot.rs
expression: expansion
---
#![feature(prelude_import)]
#[prelude_import]
use std::prelude::rust_2021::*;
#[macro_use]
extern crate std;
use std::path::Path;
use fixtures::fixtures;
fn test(_path: &Path) {}
mod test {
    use super::*;
    pub fn _abstract() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/abstract",
            ),
        )
    }
    pub fn _impl() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/impl",
            ),
        )
    }
    pub fn _move() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/move",
            ),
        )
    }
    pub fn _unsized() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/unsized",
            ),
        )
    }
    pub fn _ref() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/ref",
            ),
        )
    }
    pub fn _box() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/box",
            ),
        )
    }
    pub fn _override() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/override",
            ),
        )
    }
    pub fn _do() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/do",
            ),
        )
    }
    pub fn _match() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/match",
            ),
        )
    }
    pub fn _in() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/in",
            ),
        )
    }
    pub fn _else() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/else",
            ),
        )
    }
    pub fn _enum() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/enum",
            ),
        )
    }
    pub fn _if() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/if",
            ),
        )
    }
    pub fn _async() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/async",
            ),
        )
    }
    pub fn _return() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/return",
            ),
        )
    }
    pub fn _gen() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/gen",
            ),
        )
    }
    pub fn __() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/_",
            ),
        )
    }
    pub fn _dyn() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/dyn",
            ),
        )
    }
    pub fn _struct() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/struct",
            ),
        )
    }
    pub fn _use() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/use",
            ),
        )
    }
    pub fn _const() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/const",
            ),
        )
    }
    pub fn _await() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/await",
            ),
        )
    }
    pub fn _type() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/type",
            ),
        )
    }
    pub fn _priv() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/priv",
            ),
        )
    }
    pub fn _0_numeric_prefix_dot_txt() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/0_numeric_prefix.txt",
            ),
        )
    }
    pub fn _crate() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/crate",
            ),
        )
    }
    pub fn _macro() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/macro",
            ),
        )
    }
    pub fn _super() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/super",
            ),
        )
    }
    pub fn _self() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/self",
            ),
        )
    }
    pub fn _unsafe() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/unsafe",
            ),
        )
    }
    pub fn _static() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/static",
            ),
        )
    }
    pub fn _let() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/let",
            ),
        )
    }
    pub fn _virtual() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/virtual",
            ),
        )
    }
    pub fn _trait() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/trait",
            ),
        )
    }
    pub fn _for() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/for",
            ),
        )
    }
    pub fn _mut() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/mut",
            ),
        )
    }
    pub fn _try() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/try",
            ),
        )
    }
    pub fn _yield() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/yield",
            ),
        )
    }
    pub fn _as() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/as",
            ),
        )
    }
    pub fn _become() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/become",
            ),
        )
    }
    pub fn _final() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/final",
            ),
        )
    }
    pub fn _fn() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/fn",
            ),
        )
    }
    pub fn _pub() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/pub",
            ),
        )
    }
    pub fn _true() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/true",
            ),
        )
    }
    pub fn _break() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/break",
            ),
        )
    }
    pub fn _extern() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/extern",
            ),
        )
    }
    pub fn _where() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/where",
            ),
        )
    }
    pub fn _continue() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/continue",
            ),
        )
    }
    pub fn _false() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/false",
            ),
        )
    }
    pub fn _while() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/while",
            ),
        )
    }
    pub fn _mod() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/mod",
            ),
        )
    }
    pub fn _loop() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/loop",
            ),
        )
    }
    pub fn _typeof() {
        test(
            ::std::path::Path::new(
                "<repo>/tests/fixtures/invalid_identifiers/typeof",
            ),
        )
    }
    pub const EXPANSIONS: &[fn()] = &[
        _abstract,
        _impl,
        _move,
        _unsized,
        _ref,
        _box,
        _override,
        _do,
        _match,
        _in,
        _else,
        _enum,
        _if,
        _async,
        _return,
        _gen,
        __,
        _dyn,
        _struct,
        _use,
        _const,
        _await,
        _type,
        _priv,
        _0_numeric_prefix_dot_txt,
        _crate,
        _macro,
        _super,
        _self,
        _unsafe,
        _static,
        _let,
        _virtual,
        _trait,
        _for,
        _mut,
        _try,
        _yield,
        _as,
        _become,
        _final,
        _fn,
        _pub,
        _true,
        _break,
        _extern,
        _where,
        _continue,
        _false,
        _while,
        _mod,
        _loop,
        _typeof,
    ];
}
#[allow(dead_code)]
fn main() {}
#[rustc_main]
#[coverage(off)]
#[doc(hidden)]
pub fn main() -> () {
    extern crate test;
    test::test_main_static(&[])
}
